cdap-metrics-sidecar/src/main/java/io/cdap/cdap/jmx/metrics/JMXMetricsCollector.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.jmx.metrics;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.AbstractScheduledService;
+import com.google.inject.Inject;
+import com.sun.management.OperatingSystemMXBean;
```

Comment:
Why using this internal package instead of the `java.lang.management. OperatingSystemMXBean`?

Reply:
`java.lang.management.OperatingSystemMXBean` doesn't contain a function to get the CPU load in the in a bounded range (eg: [0, 1]). It does have a function for [Average System Load](https://docs.oracle.com/javase/7/docs/api/java/lang/management/OperatingSystemMXBean.html#getSystemLoadAverage()), but it's value isn't bounded. 
Is there a reason to avoid this package? If so, I can look into achieving the same behaviour without it.

Reply:
Generally avoid using any `com.sun` packages. As 

1. They might not be available in different Java implementations.
2. They might not be available in different Java versions.

Reply:
Understood. Replaced `com.sun` package with `java.lang.management.OperatingSystemMXBean` and ProcessCPULoad metric with AverageSystemLoad / number of processors. Multiplied metric with 100 to scale, as CDAP gauge metrics cannot store floats.